reactor-test/src/main/java/reactor/test/DefaultStepVerifierBuilder.java

```diff
@@ -1307,7 +1307,7 @@ boolean onCollect(Signal<T> actualSignal) {
 				this.completeLatch.countDown();
 				return true;
 			}
-			return true;
+			return false;
```

Comment:
is that part of the fix? can you expand on this change? it seems odd, as now the method always returns `true` (and thus the line 1343 would seem unreachable?)

Reply:
Just checked that the change in line 1310 is the only one needed.
Indeed, now `onCollect` always returns true and this [part of code](https://github.com/reactor/reactor-core/blob/master/reactor-test/src/main/java/reactor/test/DefaultStepVerifierBuilder.java#L1361-L1382) is never reached. 
The problem only occures when `expectRecordedMatches` consumes all signals from upstream. 
That is why `onComplete` makes a difference.

I also added `testExpectRecordedMatchesTwice` as I do not understand why [this is needed](https://github.com/reactor/reactor-core/blob/master/reactor-test/src/main/java/reactor/test/DefaultStepVerifierBuilder.java#L1372-L1376)